### PR TITLE
CompHEP interface to Pythia8

### PR DESCRIPTION
**Pull request by alexeybaskakov**

### CODE REVIEW

**Comment by govoni** [comment]
+1

**Comment by covarell** [comment]
please test

**Comment by covarell** [comment]
+1

**Comment by davidlange6** [comment]
Hi @alexeybaskakov @covarell - could you say more about the use of this? If its for production, the filtering and the ntupling functionality need to be separated into two modules (a filter and an analyzer).

**Comment by covarell** [comment]
So I guess the ntupling functionality can be disabled just with a switch (writeFile = false). We would use always this option in production. Would it be enough? @alexeybaskakov

**Comment by davidlange6** [comment]
it will be less work in the long term if its two modules sharing a common class to do the selection. but at this moment a switch is ok too (but it should disable the TFile and TTree creation and filling)

**Comment by alexeybaskakov** [comment]
Hi @covarell  @davidlange6 .  Yes, this filter is used for production. I will try too separate this functions and update filter.

**Comment by covarell** [comment]
+1

File: GeneratorInterface/GenFilters/src/ComphepSingletopFilterPy8.cc [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+//vars for b from top
+GenVertex * gv_b_t= NULL;
+const GenParticle * gp_b_t= NULL;
+vector<const GenParticle *> vgp_b_t;
+
+
+//Run through all particles in myEvt, if lepton found saves it gp_clep
+    for (GenEvent::particle_const_iterator it = myEvt->particles_begin(); it != myEvt->particles_end(); ++it)
+   {
+       cout <<"###  Filter Start 1 ###"<< abs((*it)->status()) <<endl;
```
Comment (davidlange6) [inline GeneratorInterface/GenFilters/src/ComphepSingletopFilterPy8.cc]: @alexeybaskakov -please remove this printout. Thanks

**Comment by covarell** [comment]
+1

**Comment by covarell** [comment]
please test

**Comment by mkirsano** [comment]
Why is it called interface if it is just a filter?

**Comment by mkirsano** [comment]
Please note that my question above is just a question, not an issue. In addition, I have already discussed this a bit, it seems that there are reasons to call it like this.

**Comment by davidlange6** [comment]
+1

**Pull request merged**